### PR TITLE
Feature: object validations

### PR DESCRIPTION
**Pull request by markvaldy**
Added handling of validations as object instead of NSDictionary.

### CODE REVIEW

File: Source/Models/FORMField.m [hunk 1]
```diff
@@ -224,9 +231,9 @@ - (FORMValidationResultType)validate
 {
     id validator;
     Class validatorClass;
-
+
```
Comment (NSElvis) [inline Source/Models/FORMField.m]: Trailing spaces are the worst, please consider using Marvin, I know @attila-at-hyper, @zenangst and me use it (it's available on Alcatraz).

https://github.com/zenangst/MarvinXcode/issues/13

File: Source/Models/FORMFieldValidation.h [hunk 2]
```diff
@@ -4,10 +4,12 @@
 @interface FORMFieldValidation : NSObject
 
 @property (nonatomic, getter = isRequired) BOOL required;
-@property (nonatomic) NSUInteger minimumLength;
-@property (nonatomic) NSInteger maximumLength;
-@property (nonatomic) NSString *format;
-@property (nonatomic) CGFloat minimumValue;
-@property (nonatomic) CGFloat maximumValue;
+@property (nonatomic, strong) NSNumber *minimumLength;
```
Comment (zenangst) [inline Source/Models/FORMFieldValidation.h]: We can remove `strong`, that's the default.

File: Source/FORMData.m [hunk 3]
```diff
@@ -315,9 +315,8 @@ - (NSMutableDictionary *)requiredFields
     for (FORMGroup *form in self.forms) {
         for (FORMSection *section in form.sections) {
             for (FORMField *field in section.fields) {
-                if (field.validations) {
-                    BOOL required = [[field.validations andy_valueForKey:@"required"] boolValue];
-                    if (required) {
+                if (field.validation) {
+                    if (field.validation.isRequired) {
```
Comment (zenangst) [inline Source/FORMData.m]: Could we chain these instead of nesting them?

**Comment by NSElvis** [comment]
I like this. :)

good_job

**Comment by zenangst** [comment]
giphy-16

File: Source/Models/FORMFieldValidation.h (hunk 2 above)
Comment (NSElvis) [inline Source/Models/FORMFieldValidation.h]: I guess we haven't done that in the app yet, but I made an issue for it!

https://github.com/hyperoslo/Form/issues/230

**Comment by NSElvis** [comment]
@markvaldy Would you mind looking at the comments so I can merge this?

**Comment by markvaldy** [comment]
@NSElvis Yes, I will.

**Comment by zenangst** [comment]
@markvaldy :+1:

**Comment by markvaldy** [comment]
@NSElvis @zenangst Everything is corrected according comments.

**Comment by NSElvis** [comment]
@markvaldy fantastic!

**Pull request merged**